### ISSUE TITLE
Camera upload only works on specific path when folder is accessible in multiple ways

### ISSUE THREAD

**Issue by tylus** [owncloud/android#2464]
When Camera Upload didn't work for me, I noticed that the same folder can be addressed through two different paths on my phone. I know both paths point to the same location because I see the same pictures in both. I don't understand this exactly, but I'm guessing one path is an alias or symlink to the other. I've waited the required 15+ minutes in both cases.

### Actual behaviour
- Camera upload does not work on a specific path, when two paths are possible to refer to the same folder

### Expected behaviour
- Download works for either path
 
### Steps to reproduce

1. Pick "Camera Folder" as `/sdcard/DCIM/Camera`
2. Take a picture A
3. Wait 20 minutes

No upload.

4. Pick "Camera Folder" as `/storage/6232-3632/DCIM/Camera`
5. Take a picture B
6. Wait 20 minutes

Upload of picture B. Picture A is still not uploaded.

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)

*Sorry, I didn't try this, because I can reproduce the behavior for the two paths, so I'm quite confident it's not about the server.*

### Environment data

Android version: 8.0.0

Device model: SM-J330F

Stock or customized system: stock

ownCloud app version: 2.10.0-beta-1

ownCloud server version: Debian, 10.1.0-1+1.1

#### Web server error log
Empty.

#### ownCloud log (data/owncloud.log)
```
{"reqId":"bnZ4Oi4B4c3KudBLZv4S","level":3,"time":"2019-02-05T21:43:20+00:00","remoteAddr":"2001:a61:356b:cd01:1a5e:fff:fe97:a58b","user":"--","app":"PHP","method":"POST","url":"\/owncloud\/index.php","message":"mkdir(): Permission denied at \/var\/www\/owncloud\/lib\/private\/Setup.php#283"}
```
Huh? This has nothing to do with the issue, it's 21 days old.

#### Web server access log for step 1
Empty.

#### Web server access log for step 2
Shows access like:
```
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:00 +0100] "GET /owncloud/ocs/v2.php/cloud/capabilities?format=json HTTP/1.1" 200 5915 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:01 +0100] "GET /owncloud/ocs/v2.php/cloud/user?format=json HTTP/1.1" 200 950 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:01 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/ HTTP/1.1" 207 1101 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:02 +0100] "GET /owncloud/index.php/avatar/miracle/256 HTTP/1.1" 200 859 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:04 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/ HTTP/1.1" 207 9601 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:04 +0100] "GET /owncloud/ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.1" 200 3655 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:24 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/ HTTP/1.1" 207 1748 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:24 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/20190227_194247.jpg HTTP/1.1" 404 952 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:25 +0100] "MKCOL /owncloud/remote.php/dav/uploads/miracle/df072cb8c748da1b4c6a8fdbb3accc451551293121751 HTTP/1.1" 201 615 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:25 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/df072cb8c748da1b4c6a8fdbb3accc451551293121751/0 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:34 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/df072cb8c748da1b4c6a8fdbb3accc451551293121751/1 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:43 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/df072cb8c748da1b4c6a8fdbb3accc451551293121751/2 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:44 +0100] "MOVE /owncloud/remote.php/dav/uploads/miracle/df072cb8c748da1b4c6a8fdbb3accc451551293121751/.file HTTP/1.1" 201 757 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:45 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/20190227_194247.jpg HTTP/1.1" 207 1607 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:46 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/ HTTP/1.1" 207 1610 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:46 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/20190227_194341.jpg HTTP/1.1" 404 952 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:47 +0100] "MKCOL /owncloud/remote.php/dav/uploads/miracle/6e4d3aee74105fe64ace8545bdd2896a1551293121784 HTTP/1.1" 201 615 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:47 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/6e4d3aee74105fe64ace8545bdd2896a1551293121784/0 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:45:56 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/6e4d3aee74105fe64ace8545bdd2896a1551293121784/1 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:46:05 +0100] "PUT /owncloud/remote.php/dav/uploads/miracle/6e4d3aee74105fe64ace8545bdd2896a1551293121784/2 HTTP/1.1" 201 648 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:46:08 +0100] "MOVE /owncloud/remote.php/dav/uploads/miracle/6e4d3aee74105fe64ace8545bdd2896a1551293121784/.file HTTP/1.1" 201 757 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
2001:a61:356b:cd01:856:f1dc:1689:8e27 - miracle [27/Feb/2019:19:46:09 +0100] "PROPFIND /owncloud/remote.php/dav/files/miracle/CameraUpload/20190227_194341.jpg HTTP/1.1" 207 1607 "-" "Mozilla/5.0 (Android) ownCloud-android/2.10.0-beta.1"
```

**Comment by davigonz** [comment]
Hi @tylus , thanks for reporting this. I see you are using a beta version, can you please attach here some logs from the Android app? The Logs option is accessible via Settings menu.

**Comment by tylus** [comment]
Hi @davigonz , sorry for the delay, and sorry for a red herring. Looking at the logs was helpful. I was previously looking at Samsung "Files" app, showing the pictures to be at

SD Card > DCIM > Camera

That is why in Owncloud, I picked `/sdcard/DCIM/Camera`. Turns out that was wrong, and it's not same folder at all.
Using Total Commander app, I found one folder to contain pictures and the other to be empty. I noticed that Total Commander thinks that "SD Card" is indeed `/storage/6232-3632`, which explains that when choosing `/storage/6232-3632/DCIM/Camera` in Owncloud, upload works.
In Owncloud, both "Camera" folders only showed an empty subfolder ".AutoPortrait", and since "Files" showed it as "SD Card > DCIM > Camera", I thought I was at the right place.

My bad. Sorry for the noise. I can't even come up with a suggestion to make this better in the UI, unless it were possible to use the pre-installed file browser to avoid different path syntax, or better: read the camera app settings.
